src/Simulation/Native/src/simulator/wavefunction.hpp

```diff
@@ -90,6 +90,53 @@ class Wavefunction
 
     void flush() const
     {
+        //logic to reorder
+        Fusion fg = fused_.getFusedGates();
+        auto items = fg.getItems();
```

Comment:
This is extremely cryptic code that is hard to read: no type information and names are generic and use inconsistent plurals. For example, what does fg.getSet() return and why it's set*s*?